tools/bin/ci_check_dependency.py

```diff
@@ -73,19 +76,45 @@ def find_file(name, path):
             return os.path.join(root, name)
 
 
-def get_depended_connectors(changed_connectors, all_build_gradle_files):
+def get_depended_connectors(changed_modules, all_build_gradle_files):
     depended_connectors = []
-    for changed_connector in changed_connectors:
+    for changed_module in changed_modules:
         for connector, gradle_file in all_build_gradle_files.items():
             with open(gradle_file) as file:
-                if changed_connector in file.read():
+                if changed_module in file.read():
                     depended_connectors.append(connector)
     return depended_connectors
 
 
 def write_report(depended_connectors):
+    affected_sources = []
+    affected_destinations = []
     for depended_connector in depended_connectors:
-        print("- " + depended_connector)
+        if depended_connector.startswith("source"):
+            affected_sources.append(depended_connector)
+        else:
+            affected_destinations.append(depended_connector)
+
+    with open(".github/connector_dependency_comment_template.md", "r") as f:
+        template = f.read()
+
+    sources_md = ""
+    for source in affected_sources:
+        sources_md += f"- {source}\n"
+
+    destinations_md = ""
+    for destination in affected_destinations:
+        destinations_md += f"- {destination}\n"
+
+    comment = template.format(
+        sources=sources_md,
+        destinations=destinations_md,
+        num_sources=len(affected_sources),
+        num_destinations=len(affected_destinations)
+    )
+
+    with open("comment_body.md", "w") as f:
+        f.write(comment)
```

Comment:
Do you have any worry about this file getting checked in?  Should it be git-ignored, or placed in `/tmp/*`?

Reply:
In this case I figured it was fine since it is only run in CI, unless you're developing on the action itself and therefore running the script manually. You can see something similar which happens [here](https://github.com/airbytehq/airbyte/pull/18480/files#diff-75feea4f8f499b045ee3a0546a8951c2f3e3f3dfef0571b4a7c5f78e63ce9143R16) -- `changed_files.txt` is created by the CI and not checked in. It doesn't exist on the branch, just in the EC2 instance (that's what we use right?) running the GHA.